packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.tsx

```diff
@@ -801,6 +783,45 @@ export class ContextualMenu extends BaseComponent<IContextualMenuProps, IContext
     }
   }
 
+  /**
+   * Hnadles updating focus when mouseEnter or mouseMove fire.
+   * As part of updating focus, This function will also update
+   * the expand/collapse state accordingly.
+   */
+  private _updateFocusOnMouseEvent(item: any, ev: React.MouseEvent<HTMLElement>) {
+    const targetElement = ev.currentTarget as HTMLElement;
+
+    if (item.key === this.state.expandedMenuItemKey) {
+      return;
+    }
+
+    if (this._enterTimerId !== undefined) {
+      this._async.clearTimeout(this._enterTimerId);
+      this._enterTimerId = undefined;
+    }
+
+    const menuExpanded = this.state.expandedMenuItemKey !== undefined;
+
+    // If the menu is not expanded we can update focus without any delay
+    if (!menuExpanded) {
+      targetElement.focus();
+    }
+
+    // Delay updating expanding/dismissing the submenu
+    // and only set focus if we have not already done so
+    if (hasSubmenu(item)) {
```

Comment:
So that I understand, this just means that the item has a submenu but not that it is expanded no? why do we treat it as menu is expanded?

Reply:
Correct, at this point we know that the item that the event fired against is not also expanded from the first if check. At this point if it has a submenu, we need to queue up an expand, otherwise queue up dismissing whatever submenu is open (which no-ops if there is no menu currently available)

Reply:
Also note, updating focus and expanding on hover is the desired behavior that follows windows menu behavior

Reply:
I see, thanks